src/core/IO/NativeDescriptor.pm6

```diff
@@ -0,0 +1,147 @@
+# On Windows, IO::NativeDescriptor represents a pointer to a HANDLE.
+# On *NIX, IO::NativeDescriptor represents a file descriptor.
+
+my class IO::NativeDescriptor is Int does IO {
+    multi method WHICH(IO::NativeDescriptor:D: --> ValueObjAt) {
+        nqp::box_s(
+          nqp::concat(
+            nqp::if(
+              nqp::eqaddr(self.WHAT, IO::NativeDescriptor),
+              'IO::NativeDescriptor|',
+              nqp::concat(self.^name, '|')
+            ),
+            nqp::coerce_is(nqp::unbox_i(self))
+          ),
+          ValueObjAt
+        )
+    }
+
+    multi method ACCEPTS(IO::NativeDescriptor:D: int \other --> Bool:D)   {
+        nqp::hllbool(nqp::iseq_i(nqp::unbox_i(self), other))
+    }
+    multi method ACCEPTS(IO::NativeDescriptor:D: Int:D \other --> Bool:D) {
+        nqp::hllbool(nqp::iseq_i(nqp::unbox_i(self), nqp::unbox_i(other)))
+    }
+
+    proto method new(IO::NativeDescriptor: |)         {*}
+    multi method new(IO::NativeDescriptor: int $fd)   {
+        nqp::box_i($fd, IO::NativeDescriptor)
+    }
+    multi method new(IO::NativeDescriptor: Int() $fd) {
+        nqp::box_i(nqp::decont_i($fd), IO::NativeDescriptor)
+    }
+
+    method Int(IO::NativeDescriptor:D: --> Int:D) {
+        nqp::p6box_i(nqp::unbox_i(self))
+    }
+
+    method IO(IO::NativeDescriptor:D: --> IO::NativeDescriptor:D) { self }
+
+    multi method Str (IO::NativeDescriptor:D: --> Str:D) {
+        nqp::p6box_s(nqp::coerce_is(nqp::unbox_i(self)))
+    }
+    multi method gist(IO::NativeDescriptor:D: --> Str:D) {
+        nqp::p6box_s(
+          nqp::concat(
+            nqp::coerce_is(nqp::unbox_i(self)),
+            '.IO'
+          )
+        )
+    }
+    multi method perl(IO::NativeDescriptor:D: --> Str:D) {
+        nqp::p6box_s(
+          nqp::concat(
+            nqp::concat(self.^name, '.new('),
+            nqp::concat(nqp::coerce_is(nqp::unbox_i(self)), ')')
+          )
+        )
+    }
+
+    multi method e       (IO::NativeDescriptor:D: --> Bool)    {
```

Comment:
This indentation style isn't used anywhere else. Adding any new method that has a longer name will require breaking the `blame` for all methods when they need to be reindented.

Reply:
I love vertical alignment but not as much :) Looks weird indeed.